File: SuperiorPackGroup/DAL/SecurityDAL.vb
Imports SuperiorPackGroup.SPGTableAdapters

Partial Public Class SecurityDAL

    Private m_SecurityTableAdapter As SecurityTableAdapter
    Public ReadOnly Property Adapter() As SecurityTableAdapter

        Get
            If m_SecurityTableAdapter Is Nothing Then
                m_SecurityTableAdapter = New SecurityTableAdapter
                m_SecurityTableAdapter.Connection.ConnectionString = My.Settings.UserConnectionString
            End If
            Return m_SecurityTableAdapter
        End Get

    End Property

    Public Function GetSecuritySettingsByUserName(ByVal userName As String) As Data.DataTableReader

        Return Adapter.GetSecuritySettingsByUserName(userName).CreateDataReader

    End Function

End Class

File: SuperiorPackGroup/DAL/SecurityDAL.cs
//INSTANT C# NOTE: Formerly VB project-level imports:
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Drawing;
using System.Diagnostics;
using System.Windows.Forms;
using SuperiorPackGroup.SPGTableAdapters;

namespace SuperiorPackGroup
{
	public partial class SecurityDAL
	{

		private SecurityTableAdapter m_SecurityTableAdapter;
		public SecurityTableAdapter Adapter
		{

			get
			{
				if (m_SecurityTableAdapter == null)
				{
					m_SecurityTableAdapter = new SecurityTableAdapter();
					m_SecurityTableAdapter.Connection.ConnectionString = Properties.Settings.Default.UserConnectionString;
				}
				return m_SecurityTableAdapter;
			}

		}

		public DataTableReader GetSecuritySettingsByUserName(string userName)
		{

			return Adapter.GetSecuritySettingsByUserName(userName).CreateDataReader();

		}

	}

}